include/fastdds/rtps/common/CacheChange.h

```diff
@@ -149,22 +152,26 @@ struct RTPS_DllAPI CacheChange_t
         serializedPayload.encapsulation = ch_ptr->serializedPayload.encapsulation;
 
         setFragmentSize(ch_ptr->fragment_size_);
-        dataFragments_->assign(ch_ptr->dataFragments_->begin(), ch_ptr->dataFragments_->end());
+        if (ch_ptr->fragment_size_ > 0)
+        {
+            dataFragments_->assign(ch_ptr->dataFragments_->begin(), ch_ptr->dataFragments_->end());
+        }
 
         isRead = ch_ptr->isRead;
     }
 
     ~CacheChange_t()
     {
-        if (dataFragments_)
+        if (dataFragments_ != nullptr)
```

Comment:
This check is unnecessary, `delete nullptr;` is a non-op.

Reply:
dataFragments_ has been removed